File: frontend/src/types/grocery.ts
export interface GroceryItem {
  name: string;
  unit: string;
  total_quantity: number;
  price_per_unit: number;
  category: string;
}

export interface GroceryList {
  plan_id: string;
  items: GroceryItem[];
  total_cost?: number | null;
}

export interface Retailer {
  id: string;
  name: string;
  available: boolean;
  logo_url?: string;
} 

File: backend/app/models/grocery.py
from pydantic import BaseModel, Field
from typing import Optional

class GroceryItem(BaseModel):
    name: str
    unit: str
    quantity: float = Field(..., ge=0)
    price_per_unit: Optional[float] = None
    cost: Optional[float] = None
    store_place_id: Optional[str] = None

class GroceryListResponse(BaseModel):
    plan_id: str
    items: list[GroceryItem]
    total_cost: Optional[float] = None 